src/baseline_automatic_pipeline_system/parse_okr_v2.py

```diff
@@ -0,0 +1,272 @@
+        * traverse all ArgumentMentions, and replace parent-mention-id and parent-id through the global mapping.
+        * modify the template - replace original single-sentence template by changing the single-sentence symbols to their cluster (Entity\Proposition) ID.
+"""
+def generate_okr_info(sentences, all_entity_mentions, all_proposition_mentions, entities, propositions):
+    from constants import MentionType
+    okr_info = {}   # initialize all keyword arguments here for okr object initialization
+    okr_info["name"] = "default_name"   #TODO how (or whether) should we decide this?
+    okr_info["sentences"] = sentences
+    okr_info["ignored_indices"] = None  # TODO does this have meaning in non-annotated okr?
+    okr_info["tweet_ids"] = None    # TODO necessary?
+
+    # when generating the EntityMention & PropositionMention objects, maintain a global mapping
+    # between global_mention_id (sent_id + "_" + mention_symbol) to mention-object
+    global_mention_id_to_mention_object = {}
+
+    # generate Entities
+    okr_info["entities"] = {}
+    for entity_id, entity in enumerate(entities, start=1):
+        entity_id = "E" + str(entity_id)
+        entity_mentions = {}
+        for new_mention_id, (mention_global_id, _) in enumerate(entity, start=1):
+            mention_info = all_entity_mentions[mention_global_id]
+            mention_object = okr.EntityMention(id=new_mention_id,
+                                               sentence_id=mention_info["sentence_id"],
+                                               indices=mention_info["indices"],
+                                               terms=mention_info["terms"],
+                                               parent=entity_id)
+
+            # add this EntityMention object to Entity.mentions dict
+            entity_mentions[new_mention_id] = mention_object
+            # maintain global mapping
+            global_mention_id_to_mention_object[mention_global_id] = mention_object
+
+        # add this entity as Entity object to okr.entities
+        all_entity_terms = [ mention.terms for mention in entity_mentions.values() ]
+        okr_info["entities"][entity_id] = okr.Entity(id=entity_id,
+                                                     name=all_entity_terms[0],
+                                                     mentions=entity_mentions,
+                                                     terms=all_entity_terms,
+                                                     entailment_graph=None)
+
+    # generate Propositions
+    okr_info["propositions"] = {}
+    for prop_id, prop in enumerate(propositions, start=1):
+        prop_id = "P" + str(prop_id)
+        prop_mentions = {}
+        for new_mention_id, (mention_global_id, _) in enumerate(prop, start=1):
+
+            # retrieve original prop-mention information by the unique id
+            mention = all_proposition_mentions[mention_global_id]
+
+            def generate_argument_mentions():
```

Comment:
I prefer not to use nested functions, I thinks it makes reading and understanding the code harder than it should be. 
Is there a way around this? Maybe use a class?